src/wallet/gtest/test_wallet_zkeys.cpp

```diff
@@ -121,9 +121,7 @@ TEST(wallet_zkeys_tests, store_and_load_zkeys) {
     ASSERT_EQ(0, addrs.size());
 
     // wallet should have one key
-    auto address = wallet.GenerateNewZKey();
-    ASSERT_NE(boost::get<libzcash::SproutPaymentAddress>(&address), nullptr);
-    auto addr = boost::get<libzcash::SproutPaymentAddress>(address);
+    auto addr = wallet.GenerateNewSproutZKey();
```

Comment:
Shouldn't we still have the null assertion? The name change in this function does not ensure this check is done automatically for us.

Reply:
see below